contrib/hadoop-store-builder/src/java/voldemort/store/readonly/mr/azkaban/GobblinDistcpJob.java

```diff
@@ -0,0 +1,197 @@
+package voldemort.store.readonly.mr.azkaban;
+
+import java.util.Map;
+import java.util.List;
+import java.net.URI;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.permission.FsAction;
+import org.apache.hadoop.fs.permission.FsPermission;
+import org.apache.hadoop.mapred.JobConf;
+
+import gobblin.runtime.api.JobExecutionResult;
+import gobblin.runtime.embedded.EmbeddedGobblin;
+import gobblin.runtime.embedded.EmbeddedGobblinDistcp;
+
+import voldemort.utils.Props;
+import voldemort.utils.logging.PrefixedLogger;
+
+import azkaban.jobExecutor.AbstractJob;
+
+
+public class GobblinDistcpJob extends AbstractJob {
+    private final static String ATTR_PREFIX = "distcpConf.";
+    private String source;
+    private final String destination;
+    private final Props props;
+
+    GobblinDistcpJob(String id, String sourceHdfsCluster, String destinationVoldemortCluster, Props props) {
+        super(id, PrefixedLogger.getLogger(GobblinDistcpJob.class.getName(), destinationVoldemortCluster));
+        this.source = sourceHdfsCluster;
+        this.destination = destinationVoldemortCluster;
+        this.props = props;
+    }
+
+    public void run() throws Exception {
+        String cdnURL = pickCDN();
+
+        if (!cdnURL.isEmpty()) {
+            info("####################################");
+            info("              Distcp");
+            info("####################################");
+            info("Using CDN: " + cdnURL);
+            try {
+                String username = props.getString("env.USER", "unknownClient");
+                String pathPrefix = props.getString(VoldemortBuildAndPushJob.PUSH_CDN_PREFIX);
+                pathPrefix = pathPrefix.endsWith("/") ? pathPrefix : pathPrefix + "/";
+                // Replace original cluster with CDN, e.g. hdfs://original:9000/a/b/c => webhdfs://cdn:50070/a/b/c
+                String cdnDir = source.replaceAll(".*://.*?(?=/)", cdnURL + pathPrefix + username);
+                FileSystem cdnRootFS = getRootFS(cdnDir);
+                Path from = new Path(source);
+                Path to = new Path(cdnDir);
+
+                if (!prereqSatisfied(cdnURL)) {
+                    warn("Please add/append \"" + cdnURL + "\" to the \"other_namenodes\" attribute in your job specification.");
+                    throw new RuntimeException("\"other_namenodes\" does not contain the CDN cluster address " + cdnURL);
+                }
+
+                deleteDir(cdnRootFS, cdnDir);
+                runDistcp(from, to);
+                deleteDirOnExit(cdnRootFS, cdnDir);
+                addPermissionsToParents(cdnRootFS, cdnDir);
+                source = cdnDir;
+                info("Use CDN HDFS cluster: " + source);
+            } catch (Exception e) {
+                warn("An exception occurred during distcp: " + e.getMessage());
+                warn("Use original HDFS cluster: " + source);
```

Comment:
Saying "Use [...]" seems to imply that it is an instruction for the user (i.e.: an instruction to change a config or something like that).

If you want to communicate that a fall back behavior has been activated (but that the user doesn't need to do anything about it), then it may be more clear to say "Using" rather than "Use".